packages/maskbook/src/plugins/NFT/utils.ts

```diff
@@ -0,0 +1,33 @@
+// nftUrl can be of these forms:
+//  - https://opensea.io/assets/<address>/<token_id>
+//  - https://app.rarible.com/token/<address>:<token_id>
+//
+// TODO: discuss about the superrare url format
+
+import { parseURL } from '../../utils/utils'
+import { openseaHostname, openseaPathnameRegexMatcher, raribleHostname, rariblePathnameRegexMatcher } from './constants'
+
+function checkUrl(url: URL): boolean {
+    if (url.hostname === openseaHostname && openseaPathnameRegexMatcher.test(url.pathname)) return true
+    if (url.hostname === raribleHostname && rariblePathnameRegexMatcher.test(url.pathname)) return true
+
+    return false
+}
+
+export function getRelevantUrl(textContent: string): URL | null {
+    const urls = parseURL(textContent)
+    for (let url of urls) {
+        let _url = new URL(url)
+        if (checkUrl(_url)) return _url
+    }
+    return null
+}
+
+export function parseNftUrl(nftUrl: URL): [string, string] | any {
+    if (nftUrl.hostname === openseaHostname && openseaPathnameRegexMatcher.test(nftUrl.pathname))
+        return [RegExp.$1, RegExp.$2]
+    else if (nftUrl.hostname === raribleHostname && rariblePathnameRegexMatcher.test(nftUrl.pathname))
+        return [RegExp.$1, RegExp.$2]
```

Comment:
Hi, those `RegExp.$1` are not recommend to use. Please check out more detail at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/n

Reply:
use named group?

```ts
matched = /..../.exec(input)
if (matched) {
   return [matched.groups.address, matched.groups.tokenId]
}
```

see https://mdn.io/named-group

Reply:
https://caniuse.com/mdn-javascript_builtins_regexp_named_capture_groups

Reply:
Named groups are not supported by some browsers atm, so I think we should just use groups.

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Groups_and_Ranges.